804 - account ecocredits

haveanicedavid opened the pull request:
## Description

Closes: regen-network/regen-registry#804

Extracts ecocredits table from regen-network/regen-registry#704 into its own component + reuses in a new route to display ecocredits for any given account

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles 
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

haveanicedavid commented on web-registry/src/components/organisms/EcocreditsTable.tsx:
```diff
@@ -0,0 +1,217 @@
+import React, { useEffect, useState } from 'react';
```
most of the code here is copied over from the prior `Ecocredits.tsx` - minus the render prop to display buttons

blushi commented on web-registry/src/App.tsx:
```diff
@@ -102,8 +103,12 @@ const App: React.FC = (): JSX.Element => {
+          />
+          <Route
+            path="ecocredits/:accountId"
+            element={<ProtectedRoute component={EcocreditsForAccount} />}
```
this shouldn't be a protected route

blushi commented on web-registry/src/App.tsx:
```diff
@@ -102,8 +103,12 @@ const App: React.FC = (): JSX.Element => {
-            path="/dashboard/ecocredits"
-            element={<ProtectedRoute component={Ecocredits} />}
+            path="ecocredits/dashboard"
+            element={<ProtectedRoute component={MyEcocredits} />}
```
this one shouldn't be protected either (we only need the keplr wallet to be connected for now, and not the user to be logged in with auth0)

blushi commented on web-registry/src/pages/EcocreditsForAccount.tsx:
```diff
@@ -0,0 +1,118 @@
+              },
+            }}
+          >
+            <a
```
I believe the link should also include the `ArrowDownIcon`

blushi commented on web-registry/src/App.tsx:
```diff
@@ -102,8 +103,12 @@ const App: React.FC = (): JSX.Element => {
+            element={<ProtectedRoute component={MyEcocredits} />}
+          />
+          <Route
+            path="ecocredits/:accountId"
```
I'm also wondering whether we should add `accounts` sub-route to the path, similarly to most block explorers
```suggestion
            path="ecocredits/accounts/:accountAddress"
```

blushi commented on web-registry/src/pages/MyEcocredits.tsx:
```diff
@@ -0,0 +1,77 @@
+import type { TableCredits } from '../components/organisms';
+
+export const MyEcocredits: React.FC = () => {
+  // const walletContext = useWallet();
```
can we enable this now and get rid of the hard coded wallet?

blushi reviewed (approved):
LGTM

blushi reviewed (changes requested):
Just noticing that the current code is actually displaying all the credit batches even if the user has no corresponding balance for them, shouldn't just we display the batches with an actual user balance? cc/ @erikalogie

erikalogie commented:
Yes, it should be just the ecocredits that the user has a balance for.

On Mon, Feb 21, 2022 at 6:59 AM Marie Gauthier ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Just noticing that the current code is actually displaying all the credit
> batches even if the user has no corresponding balance for them, shouldn't
> just we display the batches with an actual user balance? cc/ @erikalogie
> <https://github.com/erikalogie>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/regen-network/regen-web/pull/849#pullrequestreview-888520967>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACARPTO6DSAU2SD547BGJ7DU4ILIJANCNFSM5N7DRY4Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

blushi commented:
@erikalogie should we just display a message or something in case the user has no ecocredit at all?

erikalogie commented:
Yes, it can look like this:
https://www.figma.com/file/seUiQM2SPt78KJiDL0Ea8m/Basic-Ecocredits-Ledger-View?node-id=1467%3A95762

On Mon, Feb 21, 2022 at 7:32 AM Marie Gauthier ***@***.***>
wrote:

> @erikalogie <https://github.com/erikalogie> should we just display a
> message or something in case the user has no ecocredit at all?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/regen-network/regen-web/pull/849#issuecomment-1046780668>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACARPTJU2GNDK7F5F4EH3Q3U4IPEBANCNFSM5N7DRY4Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

victordelval reviewed (approved):
maybe `EcocreditsByAccount.tsx` instead?

haveanicedavid commented:
Sorry I missed these change requests @blushi - should now be fixed.

added fallback message: 
<img width="1788" alt="image" src="https://user-images.githubusercontent.com/9052511/155912571-95f0eec1-b61f-472d-8ba7-8341288b06df.png">

Also refactored a bit of the fetching logic to DRY up and renamed the component (@victordelval )
